tests/unit/igraph_layout_fruchterman_reingold.c

```diff
@@ -0,0 +1,126 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.inc"
+
+void make_box(int vertices, float half_size, igraph_vector_t *bounds) {
+    for (int i = 0; i < 4; i++) {
+        igraph_vector_init(&bounds[i], vertices);
+    }
+    igraph_vector_fill(&bounds[0], -half_size);
+    igraph_vector_fill(&bounds[1], half_size);
+    igraph_vector_fill(&bounds[2], -half_size);
+    igraph_vector_fill(&bounds[3], half_size);
+}
+
+void destroy_bounds(igraph_vector_t *bounds) {
+    for (int i = 0; i < 4; i++) {
+        igraph_vector_destroy(&bounds[i]);
+    }
+}
+
+void check_and_destroy(igraph_matrix_t *result) {
+    igraph_real_t min, max;
+    igraph_matrix_minmax(result, &min, &max);
+    IGRAPH_ASSERT(min >= -1.0);
```

Comment:
Now the checked bounds are hard-coded (box from -1..1). It would be useful to add a parameter to this function that controls that bounds that it checks. Then we can also check the case where no bounds are set, and verify that vertices didn't fly away too far, especially in the case of disconnected graphs. For this specific function, we could check that they're within e.g. a box of -50..50.

Reply:
If you have enough iterations or a high enough temperature, they are supposed to just fly away without bounds right? That is what they're doing in practice at least. I did add some -50..50 checks for the modest cases.